Source/ARTRealtimeChannel.m

```diff
@@ -588,7 +589,8 @@ - (void)emit:(ARTChannelEvent)event with:(ARTChannelStateChange *)data {
 
 - (void)transition:(ARTRealtimeChannelState)state status:(ARTStatus *)status {
     [self.logger debug:__FILE__ line:__LINE__ message:@"RT:%p C:%p (%@) channel state transitions from %tu - %@ to %tu - %@", _realtime, self, self.name, self.state_nosync, ARTRealtimeChannelStateToStr(self.state_nosync), state, ARTRealtimeChannelStateToStr(state)];
-    ARTChannelStateChange *stateChange = [[ARTChannelStateChange alloc] initWithCurrent:state previous:self.state_nosync event:(ARTChannelEvent)state reason:status.errorInfo];
+    ARTRealtimeChannelState previousState = self.state_nosync;
+    ARTChannelStateChange *stateChange = [[ARTChannelStateChange alloc] initWithCurrent:state previous:previousState event:(ARTChannelEvent)state reason:status.errorInfo];
```

Comment:
```suggestion
    ARTChannelStateChange *const stateChange = [[ARTChannelStateChange alloc] initWithCurrent:state previous:previousState event:(ARTChannelEvent)state reason:status.errorInfo];
```

Also, consider spreading the init over multiple lines for readability (given you were touching this LOC anyway).